### PR TITLE
First signup phase

### PR DESCRIPTION
**Pull request by baciu-radu**

### CODE REVIEW

File: create.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+create table patient (id integer not null, cnp varchar(255), insured boolean not null, name varchar(255), primary key (id));
+create table patient (id integer not null, cnp varchar(255), insured boolean not null, name varchar(255), primary key (id));
+create table xpatients (id integer not null, cnp varchar(255), insured boolean not null, name varchar(255), primary key (id));
```
Comment (andreimladin) [inline create.sql]: please remove this, it's not correct

File: src/main/java/us/icarenow/web/controller/form/SignUpPatientForm.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
+package us.icarenow.web.controller.form;
+
+import javax.validation.constraints.Email;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.Pattern;
+
+public class SignUpPatientForm {
+
+//    @Pattern(regexp = "//d{13}", message = "CNP invalid")
```
Comment (andreimladin) [inline src/main/java/us/icarenow/web/controller/form/SignUpPatientForm.java]: please add TODO on each commented part in order to know where to come back

File: src/main/resources/SQLQuery1.sql [hunk 3]
```diff
@@ -0,0 +1,30 @@
+
```
Comment (andreimladin) [inline src/main/resources/SQLQuery1.sql]: please delete this from here, it's not 1:1 (should have the same columns/structure) with the entities and also it's working now.
Also align with Alina about this

File: src/main/resources/patient.sql [hunk 4]
```diff
@@ -0,0 +1,8 @@
+DROP TABLE IF EXISTS patient;
```
Comment (andreimladin) [inline src/main/resources/patient.sql]: please remove this,  everything about the structure should be in the same sql (for now it's not needed and used anywhere)

File: src/main/resources/temp-application.properties [hunk 5]
```diff
@@ -0,0 +1,31 @@
+server.port=8091
```
Comment (andreimladin) [inline src/main/resources/temp-application.properties]: please remove this

File: src/main/resources/patient.sql (hunk 4 above)
Comment (baciu-radu) [reply src/main/resources/patient.sql]: remove, meaning completely delete file? or just the 1st line?

File: src/main/resources/temp-application.properties (hunk 5 above)
Comment (baciu-radu) [reply src/main/resources/temp-application.properties]: remove, meaning completely delete file? or just the 1st line?

File: src/main/java/us/icarenow/web/model/entity/Role.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package us.icarenow.web.model.entity;
+
+public enum Role {
```
Comment (AlinaSastras) [inline src/main/java/us/icarenow/web/model/entity/Role.java]: Oare asta nu poate sa fie clasa normala?
am doua motive:
motivu real: nu am inteles niciodata care e faza cu enum
motivu oficial: nu pusca cu baza de date

File: src/main/resources/SQLQuery1.sql (hunk 3 above)
Comment (AlinaSastras) [reply src/main/resources/SQLQuery1.sql]: pai daca vad bine asta e practic ce am scris eu prima data

File: src/main/java/us/icarenow/web/model/entity/User.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+@Table(name="icn_user")
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
+
+    private String email;
+
+    private String password;
```
Comment (AlinaSastras) [inline src/main/java/us/icarenow/web/model/entity/User.java]: daca am inteles bine de la Andrei si chiar trebuie sa puste denumirile cu denumirile din sql, aici cred ca o sa trebuiasca sa pui icn_password ca "password" e keyword in sql

File: src/main/java/us/icarenow/web/model/entity/Patient.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+    private int userId;
+
+    private String cnp;
+
+    private String firstName;
+
+    private String lastName;
+
+    private boolean insured;
```
Comment (AlinaSastras) [inline src/main/java/us/icarenow/web/model/entity/Patient.java]: @andreimladin eu in sql nu am boolean si am pus bit, oare daca e asa e bai?

File: src/main/resources/SQLQuery1.sql (hunk 3 above)
Comment (baciu-radu) [reply src/main/resources/SQLQuery1.sql]: @AlinaSastras da, e primul tau fisier. Il pun pe cel nou?

File: src/main/java/us/icarenow/web/model/entity/User.java (hunk 7 above)
Comment (baciu-radu) [reply src/main/java/us/icarenow/web/model/entity/User.java]: doar la password? sau si la email?

Comment (AlinaSastras) [reply src/main/java/us/icarenow/web/model/entity/User.java]: doar la password

File: src/main/resources/SQLQuery1.sql (hunk 3 above)
Comment (AlinaSastras) [reply src/main/resources/SQLQuery1.sql]: ti-l dau tie direct?
ca inca nu am apucat sa il pun nicaieri

File: src/main/java/us/icarenow/web/model/entity/User.java (hunk 7 above)
Comment (baciu-radu) [reply src/main/java/us/icarenow/web/model/entity/User.java]: Bun

File: src/main/resources/SQLQuery1.sql (hunk 3 above)
Comment (baciu-radu) [reply src/main/resources/SQLQuery1.sql]: Pune in utils pe Git pls

Comment (andreimladin) [reply src/main/resources/SQLQuery1.sql]: guys, limitati-va la taskurile voastre pe feature branches si PRs, taskul tau @radumbaciu nu are treaba cu scriptul asta, PR-ul trebuie sa fie curat

File: src/main/resources/patient.sql (hunk 4 above)
Comment (andreimladin) [reply src/main/resources/patient.sql]: yes, completely, you don't have a usage of this file, and also not in this phase (like having in memory), Alina is already working on creating the database structure

File: src/main/resources/temp-application.properties (hunk 5 above)
Comment (andreimladin) [reply src/main/resources/temp-application.properties]: completely, what do you do with this file? is this used somewhere? why would need a such "temp" file

File: src/main/java/us/icarenow/web/model/entity/Role.java (hunk 6 above)
Comment (andreimladin) [reply src/main/java/us/icarenow/web/model/entity/Role.java]: True, I remember that we did that (having Roles - enum & Role - entity), @radumbaciu Are you sure you pushed all the changes?

File: src/main/java/us/icarenow/web/model/entity/User.java (hunk 7 above)
Comment (andreimladin) [reply src/main/java/us/icarenow/web/model/entity/User.java]: nu, wait, campurile nu trebuie sa fie icn_, doar tabelele

File: src/main/java/us/icarenow/web/model/entity/Patient.java (hunk 8 above)
Comment (andreimladin) [reply src/main/java/us/icarenow/web/model/entity/Patient.java]: hai sa il lasam boolean

File: src/main/resources/SQLQuery1.sql (hunk 3 above)
Comment (adrianlibenciuc) [reply src/main/resources/SQLQuery1.sql]: @radumbaciu fyi
image

File: src/main/java/us/icarenow/web/model/entity/Role.java (hunk 6 above)
Comment (baciu-radu) [reply src/main/java/us/icarenow/web/model/entity/Role.java]: not all, just the last working & tested version. Was planning on doing the other changes in a future PR

File: src/main/java/us/icarenow/web/model/entity/User.java (hunk 7 above)
Comment (baciu-radu) [reply src/main/java/us/icarenow/web/model/entity/User.java]: ok, not changing here

File: src/main/java/us/icarenow/web/model/entity/Patient.java (hunk 8 above)
Comment (baciu-radu) [reply src/main/java/us/icarenow/web/model/entity/Patient.java]: ok

File: src/main/java/us/icarenow/web/model/entity/Role.java (hunk 6 above)
Comment (andreimladin) [reply src/main/java/us/icarenow/web/model/entity/Role.java]: Then please wait for Adi's changes. Once his changes are in main branch then update your branchand adapt to his changes

File: src/main/java/us/icarenow/web/model/entity/User.java (hunk 7 above)
Comment (andreimladin) [reply src/main/java/us/icarenow/web/model/entity/User.java]: please wait for Adi's changes. Once his changes are in main branch then update your branchand adapt to his changes

File: src/main/java/us/icarenow/web/repository/UserRepository.java [hunk 9]
```diff
@@ -0,0 +1,9 @@
+package us.icarenow.web.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
```
Comment (andreimladin) [inline src/main/java/us/icarenow/web/repository/UserRepository.java]: please wait for Adi's changes. Once his changes are in main branch then update your branchand adapt to his changes

File: src/main/java/us/icarenow/web/repository/UserRepository.java [hunk 10]
```diff
@@ -0,0 +1,9 @@
...
+import org.springframework.stereotype.Repository;
+import us.icarenow.web.model.entity.User;
+
+@Repository
```
Comment (andreimladin) [inline src/main/java/us/icarenow/web/repository/UserRepository.java]: please wait for Adi's changes. Once his changes are in main branch then update your branchand adapt to his changes

File: src/main/java/us/icarenow/web/service/UserService.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import us.icarenow.web.controller.form.SignUpPatientForm;
+import us.icarenow.web.model.entity.Role;
+import us.icarenow.web.model.entity.User;
+import us.icarenow.web.repository.UserRepository;
+
+@Service
+public class UserService {
```
Comment (andreimladin) [inline src/main/java/us/icarenow/web/service/UserService.java]: Then please wait for Adi's changes. Once his changes are in main branch then update your branchand adapt to his changes

**Review by andreimladin — CHANGES_REQUESTED** [review]
This shouldnt be here. As far as I remember I asked Alina to create a pr with her changes

File: src/main/java/us/icarenow/web/model/entity/User.java [hunk 12]
```diff
@@ -0,0 +1,52 @@
+package us.icarenow.web.model.entity;
+
+import javax.persistence.*;
+import java.util.List;
+
+@Entity
+@Table(name="icn_user")
+public class User {
+
+    @Id
```
Comment (andreimladin) [inline src/main/java/us/icarenow/web/model/entity/User.java]: please wait for Adi's changes. Once his changes are in main branch then update your branchand adapt to his changes

**Review by andreimladin — CHANGES_REQUESTED** [review]

File: src/main/resources/SQLQuery1.sql [hunk 13]
```diff
@@ -0,0 +1,91 @@
+CREATE database iCareNow;
+
+USE iCareNow;
+
+DROP TABLE IF EXISTS icn_role
```
Comment (andreimladin) [inline src/main/resources/SQLQuery1.sql]: As far as i remember i asked alina to create a new branch and pr on this changes. Please remove and ask her to add it separately

Comment (andreimladin) [inline src/main/resources/SQLQuery1.sql]: As far as i remember i asked alina to create a new branch and pr on this changes. Please remove and ask her to add it separately

**Review by andreimladin — CHANGES_REQUESTED** [review]

**Review by AlinaSastras — DISMISSED** [review]

File: pom.xml [hunk 14]
```diff
@@ -7,7 +7,7 @@
     <parent>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
-        <version>2.6.7</version>
+        <version>2.2.6.RELEASE</version>
```
Comment (adrianlibenciuc) [inline pom.xml]: to change to version 2.6.7

**Review by andreimladin — APPROVED** [review]

**Review by adrianlibenciuc — APPROVED** [review]

**Review by AlinaSastras — APPROVED** [review]

**Pull request merged**